CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc

```diff
@@ -0,0 +1,855 @@
+  float y;
+  float charge;
+  int size_x;
+  int size_y;
+  int maxPixelCol;
+  int maxPixelRow;
+  int minPixelCol;
+  int minPixelRow;
+};
+struct Rechit {
+  float x;
+  float y;
+};
+
+class SiPixelLorentzAnglePCLWorker : public DQMGlobalEDAnalyzer<SiPixelLorentzAngleCalibrationHistograms> {
+public:
+  explicit SiPixelLorentzAnglePCLWorker(const edm::ParameterSet&);
+  ~SiPixelLorentzAnglePCLWorker() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void bookHistograms(DQMStore::IBooker&,
+                      edm::Run const&,
+                      edm::EventSetup const&,
+                      SiPixelLorentzAngleCalibrationHistograms&) const override;
+
+  void dqmAnalyze(edm::Event const&,
+                  edm::EventSetup const&,
+                  SiPixelLorentzAngleCalibrationHistograms const&) const override;
+
+  void dqmBeginRun(edm::Run const&, edm::EventSetup const&, SiPixelLorentzAngleCalibrationHistograms&) const override;
+
+  void dqmEndRun(edm::Run const&,
+                 edm::EventSetup const&,
+                 SiPixelLorentzAngleCalibrationHistograms const&) const override;
+
+  const Pixinfo fillPix(const SiPixelCluster& LocPix, const PixelTopology* topol) const;
+  const std::pair<LocalPoint, LocalPoint> surface_deformation(const PixelTopology* topol,
+                                                              TrajectoryStateOnSurface& tsos,
+                                                              const SiPixelRecHit* recHitPix) const;
+  // ------------ member data ------------
+  std::string folder_;
+  bool notInPCL_;
+  std::string filename_;
+  std::vector<std::string> newmodulelist_;
+
+  // histogram etc
+  int hist_x_;
+  int hist_y_;
+  double min_x_;
+  double max_x_;
+  double min_y_;
+  double max_y_;
+  double width_;
+  double min_depth_;
+  double max_depth_;
+  double min_drift_;
+  double max_drift_;
+  int bufsize;
+
+  // parameters from config file
+  double ptmin_;
+  double normChi2Max_;
+  int clustSizeYMin_;
+  int clustSizeYMinL4_;
+  int clustSizeXMax_;
+  double residualMax_;
+  double clustChargeMaxPerLength_;
+  int hist_depth_;
+  int hist_drift_;
+
+  std::unique_ptr<TFile> hFile_;
+  std::unique_ptr<TTree> SiPixelLorentzAngleTreeBarrel_;
+  std::unique_ptr<TTree> SiPixelLorentzAngleTreeForward_;
+
+  // es consumes
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomEsToken_;
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> topoEsToken_;
+  edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectESProducerRcd> siPixelTemplateEsToken_;
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> topoPerEventEsToken_;
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomPerEventEsToken_;
+
+  // event consumes
+  edm::EDGetTokenT<TrajTrackAssociationCollection> t_trajTrack;
+};
+
+//
+// constants, enums and typedefs
+//
+
+// tree branches barrel
+int run_;
+long int event_;
```

Comment:
Can't these be private members of the class?

Reply:
Moving those variables into private class result in errors such as:
`assignment of member 'SiPixelLorentzAnglePCLWorker::panelF_' in read-only object` 
because function `dqmAnalyze` is set [const.](https://github.com/cms-sw/cmssw/blob/668228d3a520915d44d3bd4933e76d4a5f8a1978/DQMServices/Core/interface/DQMGlobalEDAnalyzer.h#L62-L66)

Reply:
Perhaps another DQM class interface is better suited:
https://github.com/cms-sw/cmssw/tree/master/DQMServices/Core